### PR TITLE
[libssh2] Fix deleted configfiles

### LINKED ISSUE #23505: [curl] build failure with libssh2
**Host Environment**
- OS: Windows
- Compiler: Visual Studio 2022 (also seen with VS 2019)

**To Reproduce**
`./vcpkg install curl[ssl,tool,non-http,ssh]:x86-windows`

**Failure logs** 
-Cut and paste the appropriate build messages from the console output.
```
...
Starting package 5/6: libssh2:x86-windows
Building package libssh2[core,zlib]:x86-windows...
-- Downloading https://github.com/libssh2/libssh2/archive/libssh2-1.10.0.tar.gz -> libssh2-libssh2-libssh2-1.10.0.tar.gz...
-- Extracting source C:/workspace/vcpkg/downloads/libssh2-libssh2-libssh2-1.10.0.tar.gz
-- Applying patch 0001-Fix-UWP.patch
-- Applying patch 0002-fix-macros.patch
-- Using source at C:/workspace/vcpkg/buildtrees/libssh2/src/sh2-1.10.0-3a72d86278.clean
-- Found external ninja('1.10.2').
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Building x86-windows-rel
-- Installing: C:/workspace/vcpkg/packages/libssh2_x86-windows/share/libssh2/copyright
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\Giles.Burrows\AppData\Local\vcpkg\archives\22\22c9ed5a172e31024eb3e74aa79f8cd550ad47a76f9aed90e9163d568e00c3b4.zip
Installing package libssh2[core,zlib]:x86-windows...
Elapsed time for package libssh2:x86-windows: 11.36 s
Starting package 6/6: curl:x86-windows
Building package curl[core,non-http,openssl,schannel,ssh,ssl,sspi,tool]:x86-windows...
-- Downloading https://github.com/curl/curl/archive/curl-7_81_0.tar.gz -> curl-curl-curl-7_81_0.tar.gz...
-- Extracting source C:/workspace/vcpkg/downloads/curl-curl-curl-7_81_0.tar.gz
-- Applying patch 0002_fix_uwp.patch
-- Applying patch 0005_remove_imp_suffix.patch
-- Applying patch 0012-fix-dependency-idn2.patch
-- Applying patch 0020-fix-pc-file.patch
-- Applying patch 0021-normaliz.patch
-- Applying patch 0022-deduplicate-libs.patch
-- Applying patch mbedtls-ws2_32.patch
-- Applying patch export-components.patch
-- Applying patch curl-7.81.0-ssl.patch
-- Using source at C:/workspace/vcpkg/buildtrees/curl/src/url-7_81_0-459d17a9d2.clean
-- Found external ninja('1.10.2').
-- Configuring x86-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: ninja -v
    Working Directory: C:/workspace/vcpkg/buildtrees/curl/x86-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\workspace\vcpkg\buildtrees\curl\config-x86-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:415 (vcpkg_execute_required_process)
  ports/curl/portfile.cmake:80 (vcpkg_cmake_configure)
  scripts/ports.cmake:145 (include)


Error: Building package curl:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+curl
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[curl]+Build+error
including:
  package: curl[core,non-http,openssl,schannel,ssh,ssl,sspi,tool]:x86-windows -> 7.81.0
    vcpkg-tool version: 2022-03-09-1affd32f93b299d5a907816c328ca3ededb73a7e
    vcpkg-scripts version: e6509c997 2022-03-10 (13 hours ago)

Additionally, attach any relevant sections from the log files above.
```

-Please attach any additional failure logs mentioned in the console output.
```
[1/2] "C:/workspace/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/bin/cmake.exe" -E chdir "C:/workspace/vcpkg/buildtrees/curl/x86-windows-dbg" "C:/workspace/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/bin/cmake.exe" -S "C:/workspace/vcpkg/buildtrees/curl/src/url-7_81_0-459d17a9d2.clean" "-DCMAKE_PROJECT_INCLUDE=C:/workspace/vcpkg/ports/curl/cmake-project-include.cmake" "-DUSE_NGHTTP2=OFF" "-DCURL_USE_OPENSSL=ON" "-DCURL_USE_MBEDTLS=OFF" "-DCURL_USE_LIBSSH2=ON" "-DBUILD_CURL_EXE=ON" "-DENABLE_ARES=OFF" "-DCURL_WINDOWS_SSPI=ON" "-DCURL_BROTLI=OFF" "-DCURL_USE_SCHANNEL=ON" "-DCURL_USE_SECTRANSP=OFF" "-DUSE_LIBIDN2=OFF" "-DUSE_WIN32_IDN=OFF" "-DUSE_WIN32_LDAP=OFF" "-DHTTP_ONLY=OFF" "-DCURL_DISABLE_LDAP=ON" "-DBUILD_TESTING=OFF" "-DENABLE_MANUAL=OFF" "-DCURL_STATICLIB=0" "-DCMAKE_DISABLE_FIND_PACKAGE_Perl=ON" "-DENABLE_DEBUG=ON" "-DCURL_CA_FALLBACK=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/workspace/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v143" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/workspace/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/workspace/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/workspace/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_DISABLE_SOURCE_CHANGES=ON" -G "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/workspace/vcpkg/packages/curl_x86-windows/debug"
FAILED: ../../x86-windows-dbg/CMakeCache.txt 
"C:/workspace/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/bin/cmake.exe" -E chdir "C:/workspace/vcpkg/buildtrees/curl/x86-windows-dbg" "C:/workspace/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/bin/cmake.exe" -S "C:/workspace/vcpkg/buildtrees/curl/src/url-7_81_0-459d17a9d2.clean" "-DCMAKE_PROJECT_INCLUDE=C:/workspace/vcpkg/ports/curl/cmake-project-include.cmake" "-DUSE_NGHTTP2=OFF" "-DCURL_USE_OPENSSL=ON" "-DCURL_USE_MBEDTLS=OFF" "-DCURL_USE_LIBSSH2=ON" "-DBUILD_CURL_EXE=ON" "-DENABLE_ARES=OFF" "-DCURL_WINDOWS_SSPI=ON" "-DCURL_BROTLI=OFF" "-DCURL_USE_SCHANNEL=ON" "-DCURL_USE_SECTRANSP=OFF" "-DUSE_LIBIDN2=OFF" "-DUSE_WIN32_IDN=OFF" "-DUSE_WIN32_LDAP=OFF" "-DHTTP_ONLY=OFF" "-DCURL_DISABLE_LDAP=ON" "-DBUILD_TESTING=OFF" "-DENABLE_MANUAL=OFF" "-DCURL_STATICLIB=0" "-DCMAKE_DISABLE_FIND_PACKAGE_Perl=ON" "-DENABLE_DEBUG=ON" "-DCURL_CA_FALLBACK=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/workspace/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v143" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/workspace/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/workspace/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/workspace/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_DISABLE_SOURCE_CHANGES=ON" -G "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=C:/workspace/vcpkg/packages/curl_x86-windows/debug"
-- The C compiler identification is MSVC 19.31.31104.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.31.31103/bin/Hostx64/x86/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Error at C:/workspace/vcpkg/scripts/buildsystems/vcpkg.cmake:800 (_find_package):
  Could not find a package configuration file provided by "Libssh2" with any
  of the following names:

    Libssh2Config.cmake
    libssh2-config.cmake

  Add the installation prefix of "Libssh2" to CMAKE_PREFIX_PATH or set
  "Libssh2_DIR" to a directory containing one of the above files.  If
  "Libssh2" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  C:/workspace/vcpkg/ports/curl/cmake-project-include.cmake:73 (find_package)
  CMakeLists.txt:48 (project)


-- Configuring incomplete, errors occurred!
See also "C:/workspace/vcpkg/buildtrees/curl/x86-windows-dbg/CMakeFiles/CMakeOutput.log".
[2/2] "C:/workspace/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/bin/cmake.exe" -E chdir "C:/workspace/vcpkg/buildtrees/curl/x86-windows-rel" "C:/workspace/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/bin/cmake.exe" -S "C:/workspace/vcpkg/buildtrees/curl/src/url-7_81_0-459d17a9d2.clean" "-DCMAKE_PROJECT_INCLUDE=C:/workspace/vcpkg/ports/curl/cmake-project-include.cmake" "-DUSE_NGHTTP2=OFF" "-DCURL_USE_OPENSSL=ON" "-DCURL_USE_MBEDTLS=OFF" "-DCURL_USE_LIBSSH2=ON" "-DBUILD_CURL_EXE=ON" "-DENABLE_ARES=OFF" "-DCURL_WINDOWS_SSPI=ON" "-DCURL_BROTLI=OFF" "-DCURL_USE_SCHANNEL=ON" "-DCURL_USE_SECTRANSP=OFF" "-DUSE_LIBIDN2=OFF" "-DUSE_WIN32_IDN=OFF" "-DUSE_WIN32_LDAP=OFF" "-DHTTP_ONLY=OFF" "-DCURL_DISABLE_LDAP=ON" "-DBUILD_TESTING=OFF" "-DENABLE_MANUAL=OFF" "-DCURL_STATICLIB=0" "-DCMAKE_DISABLE_FIND_PACKAGE_Perl=ON" "-DENABLE_DEBUG=ON" "-DCURL_CA_FALLBACK=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/workspace/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v143" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/workspace/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/workspace/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/workspace/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_DISABLE_SOURCE_CHANGES=ON" -G "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/workspace/vcpkg/packages/curl_x86-windows"
FAILED: ../CMakeCache.txt 
"C:/workspace/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/bin/cmake.exe" -E chdir "C:/workspace/vcpkg/buildtrees/curl/x86-windows-rel" "C:/workspace/vcpkg/downloads/tools/cmake-3.22.2-windows/cmake-3.22.2-windows-i386/bin/cmake.exe" -S "C:/workspace/vcpkg/buildtrees/curl/src/url-7_81_0-459d17a9d2.clean" "-DCMAKE_PROJECT_INCLUDE=C:/workspace/vcpkg/ports/curl/cmake-project-include.cmake" "-DUSE_NGHTTP2=OFF" "-DCURL_USE_OPENSSL=ON" "-DCURL_USE_MBEDTLS=OFF" "-DCURL_USE_LIBSSH2=ON" "-DBUILD_CURL_EXE=ON" "-DENABLE_ARES=OFF" "-DCURL_WINDOWS_SSPI=ON" "-DCURL_BROTLI=OFF" "-DCURL_USE_SCHANNEL=ON" "-DCURL_USE_SECTRANSP=OFF" "-DUSE_LIBIDN2=OFF" "-DUSE_WIN32_IDN=OFF" "-DUSE_WIN32_LDAP=OFF" "-DHTTP_ONLY=OFF" "-DCURL_DISABLE_LDAP=ON" "-DBUILD_TESTING=OFF" "-DENABLE_MANUAL=OFF" "-DCURL_STATICLIB=0" "-DCMAKE_DISABLE_FIND_PACKAGE_Perl=ON" "-DENABLE_DEBUG=ON" "-DCURL_CA_FALLBACK=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/workspace/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v143" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/workspace/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/workspace/vcpkg" "-D_VCPKG_INSTALLED_DIR=C:/workspace/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-DCMAKE_DISABLE_SOURCE_CHANGES=ON" -G "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=C:/workspace/vcpkg/packages/curl_x86-windows"
-- The C compiler identification is MSVC 19.31.31104.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.31.31103/bin/Hostx64/x86/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
CMake Error at C:/workspace/vcpkg/scripts/buildsystems/vcpkg.cmake:800 (_find_package):
  Could not find a package configuration file provided by "Libssh2" with any
  of the following names:

    Libssh2Config.cmake
    libssh2-config.cmake

  Add the installation prefix of "Libssh2" to CMAKE_PREFIX_PATH or set
  "Libssh2_DIR" to a directory containing one of the above files.  If
  "Libssh2" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  C:/workspace/vcpkg/ports/curl/cmake-project-include.cmake:73 (find_package)
  CMakeLists.txt:48 (project)


-- Configuring incomplete, errors occurred!
See also "C:/workspace/vcpkg/buildtrees/curl/x86-windows-rel/CMakeFiles/CMakeOutput.log".
ninja: build stopped: subcommand failed.
```

**Additional context**
We have overnight VCPKG build and it started to fail on 4th March which happened after changes for libssh2 were introduced:

75ae1ef84437bc457fb4db54360f88af57bb158a [libssh2] Fix various issues + add features (#23327)

### PR DESCRIPTION
**Pull request by Thomas1664**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #23505 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  unchanged

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

I decided against moving `vcpkg_cmake_config_fixup` down right above the license copy because someone might move it to top again.
I also added an explaining comment.

### CODE REVIEW

File: ports/libssh2/portfile.cmake [hunk 1]
```diff
@@ -42,7 +42,9 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
```
Comment (dg0yt) [inline ports/libssh2/portfile.cmake:43]: Doesn't this delete the debug config file?

Comment (Thomas1664) [reply ports/libssh2/portfile.cmake:43]: No. The debug file is copied to `${CURRENT_PACKAGES_DIR}/share` and appended with `-debug` before the directory is deleted.

Comment (Thomas1664) [reply ports/libssh2/portfile.cmake:42]: In case you mean pkgconfig files: I keep them now

Comment (dg0yt) [reply ports/libssh2/portfile.cmake:42]: I meant the cmake config. But it is handled by `vcpkg_cmake_config_fixup` before this line. As are the cmake config files for release.

File: ports/libssh2/portfile.cmake [hunk 2]
```diff
@@ -39,10 +39,10 @@ if (VCPKG_TARGET_IS_WINDOWS)
 endif()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/lib/pkgconfig")
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib/pkgconfig")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/share")
+# Do not delete the entire share directory as it contains the *-config.cmake files
```
Comment (dg0yt) [inline ports/libssh2/portfile.cmake:43]: So this is not just about cmake config, but about everything in `share/${PORT}`, in particular `copyright`.

Comment (dg0yt) [reply ports/libssh2/portfile.cmake:43]: Ignore. `copyright` is copied later.

Comment (Thomas1664) [reply ports/libssh2/portfile.cmake:43]: copyright doesn't matter because its copied in the end.

Comment (Thomas1664) [reply ports/libssh2/portfile.cmake:43]: There is a post build check that complains if `${CURRENT_PACKAGES_DIR}/debug/share` exists.

**Pull request merged**